erpnext/accounts/doctype/sales_invoice/sales_invoice_patch.py

```
@@ -0,0 +1,72 @@
+			linked_sales_orders.add(item.sales_order)
+	
+	if not linked_sales_orders:
+		return
+	
+	# Check each Sales Order for full billing/delivery and close if eligible
+	for sales_order_name in linked_sales_orders:
+		try:
+			sales_order = frappe.get_doc("Sales Order", sales_order_name)
+			
+			# Check if the Sales Order is fully billed and delivered (using 99.99% threshold for floating-point precision)
+			if (sales_order.per_billed >= 99.99 and 
```

Ruff (0.12.2)

Trailing whitespace

Remove trailing whitespace

(W291)